src/extension/feature-pregeneralized/src/main/java/org/geoserver/data/gen/info/GeneralizationInfosProviderImpl.java

```diff
@@ -5,14 +5,23 @@
  */
 package org.geoserver.data.gen.info;
 
-import java.io.File;
 import java.io.IOException;
+import java.io.InputStream;
 import java.net.URL;
 import java.net.URLDecoder;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.dom.DOMSource;
 import org.geoserver.platform.GeoServerExtensions;
 import org.geoserver.platform.GeoServerResourceLoader;
-import org.geoserver.platform.resource.Files;
-import org.geoserver.platform.resource.Resources;
+import org.geoserver.platform.resource.Paths;
+import org.geoserver.platform.resource.Resource;
+import org.geotools.data.gen.info.GeneralizationInfos;
+import org.geotools.util.URLs;
+import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
 
 /**
```

Comment:
`The default implementation for ...` seems misleading. The default one would be the one from GeoTools. Can you explain why this alternative implementation should be used, and more importantly, how? Not being used to this extension, I can't figure out how it should be used instead of the one in geotools.